WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -221,12 +221,13 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 }
                 break;
         }
-        // Enable GB on the site created by the "new user" on mobile
+
+        // Enable the block editor on sites created on mobile
         switch (requestCode) {
             case RequestCodes.CREATE_SITE:
                 if (data != null) {
                     int newSiteLocalID = data.getIntExtra(SitePickerActivity.KEY_LOCAL_ID, -1);
-                    SiteUtils.enableBlockEditorForNewUsers(mDispatcher, mAccountStore, mSiteStore, newSiteLocalID);
+                    SiteUtils.enableBlockEditor(mDispatcher, mSiteStore, newSiteLocalID);
```

Comment:
WDYT about moving the call closer to the site creation process, right after the site creation success here https://github.com/wordpress-mobile/WordPress-Android/blob/47b9d8091ec6625e1dac98cf658d7f667764113d/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/services/SiteCreationServiceManager.kt#L88?

The process can happen on the background as well and the UI won't get updated until the app returns to the foreground so, better to move the call in the background too?

Reply:
Fixed in 6e4623b95dd1e9648e9ac9e7deeb66f4c0c33c80

Reply:
Yes, i tried to move it there, but the SiteStore didn't have the site locally yet.
I guess we can go with the current solution and fix it along the way next week. Ideally it should be called right after the site is created in the server.

Reply:
> but the SiteStore didn't have the site locally yet

Arghh, right. Bummer.